librz/analysis/p/analysis_bf.c

```diff
@@ -7,35 +8,446 @@
 #include <rz_asm.h>
 #include <rz_analysis.h>
 
-static size_t countChar(const ut8 *buf, int len, char ch) {
-	int i;
-	for (i = 0; i < len; i++) {
-		if (buf[i] != ch) {
-			break;
-		}
-	}
-	return i;
-}
-
 static int getid(char ch) {
 	const char *keys = "[]<>+-,.";
 	const char *cidx = strchr(keys, ch);
 	return cidx ? cidx - keys + 1 : 0;
 }
 
+/* New IL uplift bf */
+#define BF_ADDR_SIZE  64
+#define BF_ALIGN_SIZE 8
+
+struct bf_stack_t {
+	ut64 stack[32];
+	int sp;
+};
+typedef struct bf_stack_t *BfStack;
+
+typedef struct bf_context_t {
+	BfStack stack;
+	ut64 op_count;
+} BfContext;
+
+static void bf_syscall_read(RzILVM vm, RzILOp op) {
+	ut8 c = getc(stdin);
+	BitVector bv = rz_il_bv_new_from_ut32(BF_ALIGN_SIZE, c);
+
+	RzILVal ptr_val = rz_il_dump_value(rz_il_hash_find_val_by_name(vm, "ptr"));
+
+	rz_il_vm_mem_store(vm, 0, ptr_val->data.bv, bv);
+	rz_il_free_value(ptr_val);
+}
+
+static void bf_syscall_write(RzILVM vm, RzILOp op) {
+	RzILVal ptr_val = rz_il_dump_value(rz_il_hash_find_val_by_name(vm, "ptr"));
+
+	BitVector bv = rz_il_vm_mem_load(vm, 0, ptr_val->data.bv);
+	ut32 c = rz_il_bv_to_ut32(bv);
+
+	rz_il_free_value(ptr_val);
+	rz_il_bv_free(bv);
+
+	putchar(c);
+}
+
+ut64 pop_astack(BfStack stack) {
+	if (stack->sp < 0) {
+		printf("Empty Stack\n");
+		return -1;
+	}
+
+	ut64 ret = stack->stack[stack->sp];
+	stack->sp -= 1;
+	return ret;
+}
+
+void push_astack(BfStack stack, ut64 id) {
```

Comment:
Where do your check that `stack->sp` isn't superior to 32?

Reply:
Haven't yet, I will add it.

Reply:
@Heersin it's this that could be used as a macro.